### PR TITLE
HOCS-3468 MPAM Local Dispatch

### PR DESCRIPTION
**Pull request by robertdeniszczyc2**

### CODE REVIEW

File: src/test/java/uk/gov/digital/ho/hocs/workflow/processes/MPAMPoApprovedLocalDispatch.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+@Deployment(resources = "processes/MPAM_PO_APPROVED_LOCAL_DISPATCH.bpmn")
+public class MPAMPoApprovedLocalDispatch {
+
+    @Rule
+    @ClassRule
+    public static TestCoverageProcessEngineRule rule = TestCoverageProcessEngineRuleBuilder.create()
+            .assertClassCoverageAtLeast(0.2)
```
Comment (LiamMacP) [inline src/test/java/uk/gov/digital/ho/hocs/workflow/processes/MPAMPoApprovedLocalDispatch.java:30]: 20% converage seems a little low to be aiming for with new BPMN files?

File: src/test/java/uk/gov/digital/ho/hocs/workflow/processes/MPAMPoApprovedMinDispatch.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+@Deployment(resources = "processes/MPAM_PO_APPROVED_MIN_DISPATCH.bpmn")
+public class MPAMPoApprovedMinDispatch {
+
+    @Rule
+    @ClassRule
+    public static TestCoverageProcessEngineRule rule = TestCoverageProcessEngineRuleBuilder.create()
+            .assertClassCoverageAtLeast(0.2)
```
Comment (LiamMacP) [inline src/test/java/uk/gov/digital/ho/hocs/workflow/processes/MPAMPoApprovedMinDispatch.java:30]: 20% converage seems a little low to be aiming for with new BPMN files?

**Review by LiamMacP — CHANGES_REQUESTED** [review]
Camunda userTask's have a `Form Key` that can be specified and used.  

image

Could become the following:

image

Which is arguably more simple. I feel that fixing forward, the service tasks should probably be removed?

File: src/test/java/uk/gov/digital/ho/hocs/workflow/processes/MPAM.java [hunk 3]
```diff
@@ -102,7 +105,7 @@ public void happyPath() {
                 .execute();
 
         verify(mpamProcess)
-                .hasCompleted("CallActivity_1c4zy60");
+                .hasCompleted("CallActivity_0g8kpaf");
```
Comment (LiamMacP) [inline src/test/java/uk/gov/digital/ho/hocs/workflow/processes/MPAM.java:108]: Worth associating this in a variable at the top, would make it clearer what this CallActivity does?

File: src/test/java/uk/gov/digital/ho/hocs/workflow/processes/MPAMPoApprovedMinDispatch.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+    @Before
+    public void setup() {
+        Mocks.register("bpmnService", bpmnService);
+    }
+
+    @Test
+    public void isValidInput_ThenEndStage() {
+
+        when(processScenario.waitsAtUserTask("Validate_UserInput"))
```
Comment (LiamMacP) [inline src/test/java/uk/gov/digital/ho/hocs/workflow/processes/MPAMPoApprovedMinDispatch.java:48]: Could get 100% coverage of this file by specifying a second `thenReturn` before with `valid` set to `false`.

Comment (robertdeniszczyc2) [reply src/test/java/uk/gov/digital/ho/hocs/workflow/processes/MPAMPoApprovedMinDispatch.java:48]: Good spot

**Comment by robertdeniszczyc2** [comment]
> Camunda userTask's have a `Form Key` that can be specified and used.
> 
> image
> 
> Could become the following:
> 
> image
> 
> Which is arguably more simple. I feel that fixing forward, the service tasks should probably be removed?

Probably need to catch up with you about this, not sure what I specify in the Form Key

**Review by LiamMacP — APPROVED** [review]
Looks good to me, worth creating a TechDebt ticket to look at back button redirecting to dashboard if crossing stage?

**Pull request merged**